lib/chef/provider/package/dnf/dnf_helper.py

```diff
@@ -98,14 +98,24 @@ def query(command):
         q = q.available()
 
     if 'epoch' in command:
-        q = q.filterm(epoch=int(command['epoch']))
+        if not dnf.util.is_glob_pattern(command['epoch']):
```

Comment:
why drop this if it's a glob? Is there not a epoch__glob filter?

Reply:
no if its a glob, it is '*' so it matches anything so you don't filter.

i beat my head against that one for a bit.

i suppose i could try a comment.

Reply:
what if it's a glob that's *not* that. what if my glob is `1?` ?

Reply:
looks like that is actually legal, but i don't know how to fix that since i don't see how the underlying dnf libraries do that globbing.

so that's just going to be known broken.  i can't imagine that is an actually useful use case though.  and the dnf libaries themselves have several bugs around handling wildcards in epoch versions (see the #9763 use cases which work and which don't).